src/main/java/dk/dtu/compute/se/pisd/roborally/model/board/Space.java

```diff
@@ -54,14 +72,71 @@
      */
     private Player player;
 
-    public Space(Board board, int x, int y, BoardElement element, Heading... walls) {
+    public Space(Board board, int x, int y, BoardElement... elements) {
         this.board = board;
         this.x = x;
         this.y = y;
         player = null;
 
-        this.element = element;
-        this.walls = walls;
+        this.elements = new ArrayList<>();
+        activationElements = new ArrayList<>();
+
+        if (elements == null) return;
+        Collections.addAll(this.elements, elements);
+
+        for (BoardElement element: elements) {
+            if (element instanceof ActivationElement) activationElements.add((ActivationElement) element);
+        }
+        activationElements.sort(Comparator.comparingInt(ActivationElement::getPriority));
+    }
+
+    /**
+     * <p>Returns an array of all {@link BoardElement}s on this
+     * space. The returned array is safe; any changes to the
+     * array will not be reflected in the {@link #elements}
+     * of this space.</p>
+     * @return a safe array of all the board elements on this space
+     */
+    public BoardElement[] getElements() {
+        if (elements == null || elements.size() < 1) return new BoardElement[0];
+        return (BoardElement[]) elements.toArray();
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [BC: Impossible downcast of toArray() result (BC_IMPOSSIBLE_DOWNCAST_OF_TOARRAY)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bc-impossible-downcast-of-toarray-result-bc-impossible-downcast-of-toarray).
This code is casting the result of calling toArray() on a collection to a type more specific than Object[], as in:
This will usually fail by throwing a ClassCastException. The toArray() of almost all collections return an Object[]. They can't really do anything else, since the Collection object has no reference to the declared generic type of the collection. The correct way to do get an array of a specific type from a collection is to use  c.toArray(new String[]);  or c.toArray(new String[c.size()]); (the latter is slightly more efficient). There is one common/known exception to this. The toArray() method of lists returned by Arrays.asList(...) will return a covariantly typed array. For example, Arrays.asArray(new String[] { "a" }).toArray() will return a String []. SpotBugs attempts to detect and suppress such cases, but may miss some.
The correct way to do get an array of a specific type from a collection is to use  c.toArray(new String[]);  or c.toArray(new String[c.size()]); (the latter is slightly more efficient). There is one common/known exception to this. The toArray() method of lists returned by Arrays.asList(...) will return a covariantly typed array. For example, Arrays.asArray(new String[] { "a" }).toArray() will return a String []. SpotBugs attempts to detect and suppress such cases, but may miss some.
There is one common/known exception to this. The toArray() method of lists returned by Arrays.asList(...) will return a covariantly typed array. For example, Arrays.asArray(new String[] { "a" }).toArray() will return a String []. SpotBugs attempts to detect and suppress such cases, but may miss some.

Reply:
Was already fixed